spec/emerald/parser_spec.rb

```diff
@@ -100,4 +100,60 @@
                           Emerald::Number.new(30)])
     end
   end
+
+  describe "string parsing" do
+    it"parses a string ofcharacters" do
+      source = '"Hello"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello"')])
+    end
+    
+    it"parses a string ofcharacters and whitespace" do
+      source = '"Hello this is a string"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello this is a string"')])
+    end
+
+    it "parses a string of characters, whitespace and digits" do
+      source = '"H3llo this is 1 str1ng"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo this is 1 str1ng"')])
+    end
+
+    it "parses a string of characters, whitespace, digits and symbols" do
+      source = '"H3llo, this is 1 str1ng."' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo, this is 1 str1ng."')])
+    end
+
+    it "parses unusual strings that contain quotation marks" do
+      source = '"this is a weird string -> \" <- "'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"this is a weird string -> \" <- "')])
+    end
+    
+    it "parses strings which contain several backslashes" do
+      source = '"This is also a \\\weird "string"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"This is also a \\\weird "string"')])
+    end
+
+    it "parses strings with symbols" do
+      source = '"!!@@££$$%%^^&&**()"\]?"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"!!@@££$$%%^^&&**()"\]?"')])
+    end
+
+    it "parses strings with double backslashes" do
+      source = '"\\"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"\\"')])
+    end
+
+    it "parses strings containing backslashes and quotation marks" do
+      source = '"\\\'""'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"\\\'""')])
+    end
+  end
```

Comment:
These two pass - \S\s in regex will account for any whitespace and non-whitespace character